### ISSUE TITLE
Large .psb files

### ISSUE THREAD

**Issue by slaymantis** [VoidVolker/MagickSlicer#8]
Hi

What is the largest we can go using this tool?
I have a 121 095 x 52 585 pixel image I am trying to convert
It seems to get stuck at level 4 of the zoom tiles

I tried smaller files and they work great

**Comment by VoidVolker** [comment]
Hi! It shall work for any image size. What is your OS and what commands you using?

**Comment by slaymantis** [comment]
Hi - it works well with my images up to about 40 000 pixels. But the 121 000 pixels it just seems to get stuck on my one computer and the other computer says terminal has run out of memory.
The command I am using is
./magick-slicer.sh new_dubai_night_flat.psb -w 512 -e jpg

The machine specs are:
OSX el capitan 32gig ram i5 iMac
OSX sierra 16 gig ram i5 iMac
OSX el capitan 16gig ram i7 macbook pro

I haven't been able to render any of them with the large image (also left the i7 machine to run for 3 days but no luck - it just had one empty folder)

**Comment by VoidVolker** [comment]
I run script and found problem with defining image sizes from imagemagick - it require at last 2-3Gb RAM and more 50Gb on disk. Try to add option `-v3` and check on what stage it stuck, I suspect there can be some error from imagemagick and it can stuck in loop with incorrect parameters.

**Comment by slaymantis** [comment]
Thanks
I see this error multiple time when I try run with or without defining the tilesize on the larger image (15GB image)
(no errors on the smaller images like 2GB image)

./magick-slicer.sh: line 665: let: imgLen = imgLen * 100 / zoomStep: division by 0 (error token is "p")

**Comment by VoidVolker** [comment]
Yes, I'm already found this error - looks like it produced by `identify`, because it can't identify image sizes correctly, because it require alot of disk space. I'll fix it when I'll have time in next few days. Thanks for bug report. If your image not secret - can you share it, please? I'll use it for tests on real image.

**Comment by slaymantis** [comment]
Thank you

The image is a client image and very very large 15GB- (I can get permission and watermark it though - do you have a private email so I can send you a link?)

Otherwise  the image is easy to make in photoshop for testing?
Just take any image and scale it up to:
121095 x 52585 pixels
72dpi

and save it as a large format photoshop file .psb

Let me know if you need a link

**Comment by VoidVolker** [comment]
Yes, I'm already done this in photoshop for tests. You can contact me directly in [gitter webchat](https://gitter.im/VoidVolker/MagickSlicer)

**Comment by srussking** [comment]
Just in case this happens to me again I wanted to mention that you can also get this error ( imgLen = imgLen * 100 / zoomStep: division by 0) if your image magick policy is set too low for the image you are trying to process. Increasing the memory limits as described here might help you. [more info](https://github.com/ImageMagick/ImageMagick/issues/396[)

**Comment by mklucz** [comment]
Hi, I was getting the following error
`./magick-slicer.sh: line 665: let: imgLen = imgLen * 100 / zoomStep: division by 0 (error token is "zoomStep")`
while trying to process a large image.
I fixed it by going to `/etc/ImageMagick-6/policy.xml` and changing the values in these two lines
`  <policy domain="resource" name="width" value="16KP"/>`
`  <policy domain="resource" name="height" value="16KP"/>`
to be greater than my image's size.
One may also need to increase allowed memory, it's in the same file.
